src/asynqp/connection.py

```diff
@@ -60,9 +60,10 @@ def close(self):
 
         This method is a :ref:`coroutine <coroutine>`.
         """
-        self.sender.send_Close(0, 'Connection closed by application', 0, 0)
-        yield from self.synchroniser.await(spec.ConnectionCloseOK)
-        self.closed.set_result(True)
+        if not self.closed.done():
+            self.sender.send_Close(0, 'Connection closed by application', 0, 0)
+            yield from self.synchroniser.await(spec.ConnectionCloseOK)
+            self.closed.set_result(True)
```

Comment:
Should, for example, `queue.get()` throw an exception if the connection is closed, or just return `None`? It seems pretty clear to me that the right choice in that case is to throw an exception, especially since `None` is a valid return value for `queue.get()` under ordinary circumstances.

I value consistency in APIs, so I think all the methods which talk to the server over the network should behave in the same way. That means throwing an exception if the connection has been closed. `connection.close()` is only special in the respect that it typically happens when the application is shutting down.